### PR TITLE
add rolemarker at the side

### LINKED ISSUE #718: feat(rolemarker): add ability to check current role still
preferably by hovering over the icon

### PR DESCRIPTION
**Pull request by pikulet**
[Screencast from 29-04-2023 01:01:08.webm](https://user-images.githubusercontent.com/24848927/235209134-6bc40f15-3392-4ca2-8020-7de059fd4e64.webm)

Fixes #718

### CODE REVIEW

File: react_main/src/components/Roles.jsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React, { useState, useContext, useRef } from "react";
 import axios from "axios";
 
-import { UserContext, SiteInfoContext, PopoverContext, GameContext } from "../Contexts";
+import { UserContext, SiteInfoContext, PopoverContext } from "../Contexts";
```
Comment (pikulet) [inline react_main/src/components/Roles.jsx:4]: removed toggleroleprediction from this file

toggleroleprediction opens the role marker pop up

File: react_main/src/pages/Game/Game.jsx [hunk 2]
```diff
@@ -1544,10 +1544,45 @@ export function SideMenu(props) {
...
+            }
+        )
+    }
+
+    return (
+        <div className="role-marker" 
+            onClick={onRoleMarkerClick}
+            ref={roleMarkerRef}>
+                
+            <i className="fas fa-map-pin"></i>
```
Comment (pikulet) [inline react_main/src/pages/Game/Game.jsx:1576]: i'm ok to change to another icon, i don't particularly like this but i'm not sure what others are good

**Comment by pikulet** [comment]
image
changed icon

**Comment by starryburst** [comment]
i was looking at this and it looks fine but a minor thing is that the ui is in front of the icon so you can't really see it changing until you close the role marker icon, it could be bothering

**Comment by pikulet** [comment]
good point, I'll want to merge this first and fix that later, because this one needs time to look at!

**Review by starryburst — APPROVED** [review]
fine besides blocking role ui as said earlier but works properly

**Review by DrSharky — APPROVED** [review]
Hahaha just wait until there's mobile bug complaints. You know it'll happen! APPROVED.

**Pull request merged**